### PR TITLE
[Android] Respect both image size and column width

### LINKED ISSUE #4439: [Android][Rendering] Conflict between image size and column weight
# Platform
* Android

# Details

With the introduction of image resizing to a bigger scale the ```v1.2/Elements/Column.Weighted.Image.json``` card became broken as there's a conflict between the image size and the column weight.

image

### PR DESCRIPTION
**Pull request by golddove**
## Related Issue
Fixes #4439 

## Description
Uses constraint layout to make images grow to the requested size, but not larger than the parent container/column.

## How Verified
Manual verification:

`v1.2/Tests/ImageSizing.json`:

`ActivityUpdate.json`:

image

`FlightItinerary.json`:
image

`ProductVideo.json`:
image
image

`Agenda.json`:
image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4613)

### CODE REVIEW

File: source/android/adaptivecards/src/main/res/layout/image_constraint_layout.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        app:layout_constraintGuide_percent="0" />
+
+    <android.support.constraint.Barrier
+        android:id="@+id/desiredLeft"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:barrierDirection="left"
+        app:constraint_referenced_ids="widthPlaceholder" />
+
+
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/res/layout/image_constraint_layout.xml:39]: nit: extra spaces

File: samples/v1.2/Tests/ImageSizing.json [hunk 2]
```diff
@@ -1,153 +1,336 @@
...
-                {
-                    "type": "Image",
-                    "url": "https://picsum.photos/id/133/300/200"
-                }
-            ]
-        }
-    ]
-}
\ No newline at end of file
+	"type": "AdaptiveCard",
```
Comment (almedina-ms) [inline samples/v1.2/Tests/ImageSizing.json:2]: nice catch 👍

**Comment by almedina-ms** [comment]
Last time there were problems with media and other elements, could you do a quick run across all cards to check that everything is working as expected?

**Comment by golddove** [comment]
Test evidence added to PR description

**Pull request merged**